File: app/src/main/java/com/earaujo/usinglivedata/rest/ApiInterface.java
package com.earaujo.usinglivedata.rest;

import com.earaujo.usinglivedata.rest.model.SearchModel;
import retrofit2.Call;
import retrofit2.http.*;

public interface ApiInterface {

    @GET("/search.json")
    Call<SearchModel> search(
            @Query("q") String search,
            @Query("limit") int limit
    );
}

File: app/src/main/java/com/earaujo/usinglivedata/rest/ApiInterface.kt
package com.earaujo.usinglivedata.rest

import com.earaujo.usinglivedata.rest.model.reddit.SearchModel
import retrofit2.Call
import retrofit2.http.*

interface ApiInterface {

    @GET("/search.json")
    fun search(
        @Query("q") search: String,
        @Query("limit") limit: Int
    ): Call<SearchModel>
}
